clientapi/routing/auth_fallback.go

```diff
@@ -0,0 +1,176 @@
+const RecaptchaTemplate = `
+<html>
+<head>
+<title>Authentication</title>
+<meta name='viewport' content='width=device-width, initial-scale=1,
+    user-scalable=no, minimum-scale=1.0, maximum-scale=1.0'>
+<script src="https://www.google.com/recaptcha/api.js"
+    async defer></script>
+<script src="//code.jquery.com/jquery-1.11.2.min.js"></script>
+<link rel="stylesheet" href="/_matrix/static/client/register/style.css">
+<script>
+function captchaDone() {
+    $('#registrationForm').submit();
+}
+</script>
+</head>
+<body>
+<form id="registrationForm" method="post" action="{{.MyUrl}}">
+    <div>
+        <p>
+        Hello! We need to prevent computer programs and other automated
+        things from creating accounts on this server.
+        </p>
+        <p>
+        Please verify that you're not a robot.
+        </p>
+		<input type="hidden" name="session" value="{{.Session}}" />
+        <div class="g-recaptcha"
+            data-sitekey="{{.SiteKey}}"
+            data-callback="captchaDone">
+        </div>
+        <noscript>
+        <input type="submit" value="All Done" />
+        </noscript>
+        </div>
+    </div>
+</form>
+</body>
+</html>
+`
+
+// SuccessTemplate is template for success page after auth flow ends
+const SuccessTemplate = `
+<html>
+<head>
+<title>Success!</title>
+<meta name='viewport' content='width=device-width, initial-scale=1,
+    user-scalable=no, minimum-scale=1.0, maximum-scale=1.0'>
+<link rel="stylesheet" href="/_matrix/static/client/register/style.css">
+<script>
+if (window.onAuthDone) {
+    window.onAuthDone();
+} else if (window.opener && window.opener.postMessage) {
+     window.opener.postMessage("authDone", "*");
+}
+</script>
+</head>
+<body>
+    <div>
+        <p>Thank you!</p>
+        <p>You may now close this window and return to the application.</p>
+    </div>
+</body>
+</html>
+`
+
+// ServeTemplate fills data in template and serves it in http.ResponseWriter
+func ServeTemplate(w http.ResponseWriter, templateHTML string, data map[string]string) {
+	t := template.Must(template.New("response").Parse(templateHTML))
+	if err := t.Execute(w, data); err != nil {
+		panic(err)
+	}
+}
+
+// AuthFallback implements GET on /auth/{authType}/fallback/web?session={sessionID}
+func AuthFallback(
+	w http.ResponseWriter, req *http.Request, authType string,
+	cfg config.Dendrite,
+) *util.JSONResponse {
+	sessionID := req.URL.Query().Get("session")
+
+	if sessionID == "" {
+		_, err := w.Write([]byte("Session ID not provided"))
+		if err != nil {
+			res := httputil.LogThenError(req, err)
+			return &res
+		}
+		return nil
+	}
+
+	ServeRecaptcha := func() {
+		data := map[string]string{
+			"MyUrl":   req.URL.String(),
+			"Session": sessionID,
+			"SiteKey": cfg.Matrix.RecaptchaPublicKey,
+		}
+		ServeTemplate(w, RecaptchaTemplate, data)
+	}
+
+	ServeSuccess := func() {
+		data := map[string]string{}
+		ServeTemplate(w, SuccessTemplate, data)
+	}
+
+	if req.Method == "GET" {
```

Comment:
```suggestion
	if req.Method == http.MethodGET {
```

Always use constants where possible.